iosapp/lib-ios/Sources/LibApp/IOSReducer.swift

```diff
@@ -0,0 +1,924 @@
+      return .none
+
+    case (.onboarding(.authFail(.invalidAccount(.confirmInAppleFamily))), .secondary):
+      self.log(state.screen, action, "39c52acf")
+      state.screen = .onboarding(.appleFamily(.explainRequiredForFiltering))
+      return .none
+
+    case (.onboarding(.authFail(.invalidAccount(.confirmInAppleFamily))), .tertiary):
+      self.log(state.screen, action, "a9cbe4fe")
+      state.screen = .onboarding(.appleFamily(.checkIfInAppleFamily))
+      state.onboarding.returningTo = .onboarding(.authFail(.invalidAccount(.confirmInAppleFamily)))
+      return .none
+
+    case (.onboarding(.authFail(.invalidAccount(.confirmIsMinor))), .primary):
+      self.log(state.screen, action, "9d0d9eac")
+      state.screen = .onboarding(.major(.explainHarderButPossible))
+      return .none
+
+    case (.onboarding(.authFail(.invalidAccount(.confirmIsMinor))), .secondary):
+      self.log(state.screen, action, "e457cf15")
+      state.screen = .onboarding(.authFail(.invalidAccount(.unexpected)))
+      return .none
+
+    case (.onboarding(.authFail(.authConflict)), .primary):
+      self.log(state.screen, action, "7b53bdc0")
+      state.screen = .onboarding(.happyPath(.explainTwoInstallSteps))
+      return .none
+
+    case (.onboarding(.authFail(.networkError)), .primary):
+      self.log(state.screen, action, "16e57d91")
+      state.screen = .onboarding(.happyPath(.explainTwoInstallSteps))
+      return .none
+
+    case (.onboarding(.authFail(.passcodeRequired)), .primary):
+      self.log(state.screen, action, "d2888470")
+      state.screen = .onboarding(.happyPath(.explainTwoInstallSteps))
+      return .none
+
+    case (.onboarding(.authFail(.authCanceled)), .primary):
+      self.log(state.screen, action, "6e3b2c93")
+      state.screen = .onboarding(.happyPath(.explainTwoInstallSteps))
+      return .none
+
+    case (.onboarding(.installFail(.permissionDenied)), .primary):
+      self.log(state.screen, action, "b122af01")
+      state.screen = .onboarding(.happyPath(.explainInstallWithDevicePasscode))
+      return .none
+
+    case (.onboarding(.installFail(.other)), .primary):
+      self.log(state.screen, action, "cf059547")
+      state.screen = .onboarding(.happyPath(.explainInstallWithDevicePasscode))
+      return .none
+
+    case (.onboarding(.childIsOnboardingFail), .primary):
+      self.log(state.screen, action, "566a3484")
+      state.screen = .onboarding(.happyPath(.hiThere))
+      return .none
+
+    case (.supervisionSuccessFirstLaunch, .primary):
+      self.log(state.screen, action, "aa563df6")
+      state.screen = .onboarding(.happyPath(.optOutBlockGroups))
+      return .none
+
+    default:
+      #if DEBUG
+        fatalError("Unhandled combination:\n -> btn: .\(btn)\n -> screen: .\(state.screen)")
+      #else
+        self.log(state.screen, action, "7c039b10", extra: "UNHANDLED ACTION")
+        state.screen = state.screen.fallbackDestination(from: btn)
```

Comment:
here is where i fatalError in debug/test but fallback in prod in case any combination is unhandled. i'm pretty confident i have every path covered by a test though currently.